### DISCUSSION TITLE
Groupby data from a different table

### DISCUSSION THREAD

**Discussion by rodbs** [refinedev/refine#5403] [Q&A] [answered]
Is it possible to query data from a linked table  to summarise it and list altogether
Example: I have users and posts . I'd like to list all users and for each user have a field that count the total number of posta this user has.

**Comment by BatuhanW** [comment]
Hello @rodbs it depends on your backend. IF you are using GraphQL, you can easily to that, but if you are using REST, and if your API doesn't support it, you will need to make 2 separate requests.

First, you can get a list of users, then map these user ids to make 2nd request with `useMany` hook.

https://refine.dev/docs/core/hooks/data/use-many/

**Comment by rodbs** [comment]
I've checked out the Strapi GraphQL example, and I see the graphql parameters are 'embedded' in the meta field? is this the way to build custom queries in GraphQL?
I opened this [#5395](https://github.com/refinedev/refine/issues/5395) because  I see with REST is  complicated to define custom queries with the `useCustom` hook ...

**Reply by BatuhanW** [reply, marked as answer]
We've revised GraphQL usage to accept GraphQL operations created with `graphql-tag` package recently for `nestjs-query` data provider, you can see it here: https://refine.dev/docs/data/packages/nestjs-query/

We also have an open PR to do the same for Hasura: https://github.com/refinedev/refine/pull/5478

Even though we accept gqlQuery, we would still need to pass filters to the data provider in use**Many hooks. 

We are open to contributions!
